cmd/collector/app/span_processor.go

```diff
@@ -87,27 +100,38 @@ func newSpanProcessor(spanWriter spanstore.Writer, opts ...Option) *spanProcesso
 	boundedQueue := queue.NewBoundedQueue(options.queueSize, droppedItemHandler)
 
 	sp := spanProcessor{
-		queue:           boundedQueue,
-		metrics:         handlerMetrics,
-		logger:          options.logger,
-		preProcessSpans: options.preProcessSpans,
-		filterSpan:      options.spanFilter,
-		sanitizer:       options.sanitizer,
-		reportBusy:      options.reportBusy,
-		numWorkers:      options.numWorkers,
-		spanWriter:      spanWriter,
-		collectorTags:   options.collectorTags,
-	}
-	sp.processSpan = ChainedProcessSpan(
-		options.preSave,
-		sp.saveSpan,
-	)
+		queue:              boundedQueue,
+		metrics:            handlerMetrics,
+		logger:             options.logger,
+		preProcessSpans:    options.preProcessSpans,
+		filterSpan:         options.spanFilter,
+		sanitizer:          options.sanitizer,
+		reportBusy:         options.reportBusy,
+		numWorkers:         options.numWorkers,
+		spanWriter:         spanWriter,
+		collectorTags:      options.collectorTags,
+		dynQueueSizeWarmup: options.dynQueueSizeWarmup,
+		stopCh:             make(chan struct{}),
+		dynQueueSizeMemory: options.dynQueueSizeMemory,
+	}
+
+	processSpanFuncs := []ProcessSpan{}
```

Comment:
can be initialized with `options.preSave, sp.saveSpan`, instead of adding them later, since they are not optional

Reply:
Would the order matter? I don't think it would, but it's worth asking...